Fix traversal limit in array metadata serialization

ypatia opened the pull request:
[sc-46026]

Reading the VCF `allele_count` array with a specific `tiledb://` URI was failing with `TileDBError: [TileDB::Serialization] Error: Error deserializing array metadata; kj::Exception: Exceeded message traversal limit.  See capnp::ReaderOptions.`

This is a classic issue we have seen before where Capnp is reaching the traversal limit and is solved by explicitly setting the traversal limit to `rest.capnp_traversal_limit` config option.

As validation , I have run the failing python script provided in the ticket description and now it passes.

---
TYPE: BUG
DESC: Fix traversal limit in array metadata serialization

gspowley reviewed (approved):
Thanks @ypatia!

robertbindar reviewed (approved):
Thanks a lot Ypatia! Was it the last code path not using the config in serialization? If not, maybe at some point we do all the ramaining ones at once, it seems it's a matter of time and user creativity until we hit one of the paths not guarded by the configurable limit :D

ypatia commented:
> Thanks a lot Ypatia! Was it the last code path not using the config in serialization? If not, maybe at some point we do all the ramaining ones at once, it seems it's a matter of time and user creativity until we hit one of the paths not guarded by the configurable limit :D

We should absolutely do that, at least audit if something is needed, Thanks for the suggestion, opened this to track : https://app.shortcut.com/tiledb-inc/story/47308/check-and-fix-other-deserialization-paths-where-capnp-traversal-limit-might-be-hit
I just fixed whatever path was using array metadata that can get big, and we had fixed before fragment metadata and array. But we could be extra cautious and just use the config in every deserialization function.

ypatia commented:
> Thanks @ypatia!

You are welcome @gspowley . As this is a client side only fix, it will be included in the next release of `libtiledb` (2.24) and no other REST Server changes are required. Should we also backport to 2.23 to unblock your work?

gspowley commented:
>  Should we also backport to 2.23 to unblock your work?

There is a workaround, so I'm not blocked. We can wait for 2.24.
